src/jit/optimizer.cpp

```diff
@@ -7977,7 +7977,7 @@ void                Compiler::optOptimizeBools()
 
             genTreeOps      foldOp;
             genTreeOps      cmpOp;
-            var_types       foldType = c1->TypeGet();
+            var_types       foldType = t1->TypeGet();
```

Comment:
I'm not sure that this fix is correct:

Consider:
JTRUE (c1 != 0) -> BX
JTRUE (c2 !=0 ) -> BX
                    /* t1:c1!=0 t2:c2!=0 ==> Branch to BX if either value is non-0
                       So we will branch to BX if (c1|c2)!=0 */
Let the values of c1 and c2 be 64-bit values with their low 342-bits as all zeros
The types of t1 and t2 are TYP_INT as these are the !=  RelOp nodes
The correct type for the GT_OR is the 64-bit type